Simplenote/Classes/SPNoteHistoryViewController.swift

```diff
@@ -0,0 +1,179 @@
+    @IBOutlet private weak var restoreButton: UIButton!
+    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
+
+    private let controller: SPNoteHistoryController
+    private var items: [SPNoteHistoryController.Presentable] = []
+
+    /// Designated initialize
+    ///
+    /// - Parameters:
+    ///     - controller: business logic controller
+    ///
+    init(controller: SPNoteHistoryController) {
+        self.controller = controller
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        styleDateLabel()
+        styleErrorMessageLabel()
+        styleSlider()
+        styleRestoreButton()
+        styleActivityIndicator()
+
+        listenForSliderValueChanges()
+
+        controller.observer = { [weak self] state in
+            self?.update(with: state)
+        }
+        controller.onViewLoad()
+
+        trackScreen()
+    }
+}
+
+// MARK: - Private Methods
+//
+private extension SPNoteHistoryViewController {
+    func styleDateLabel() {
+        dateLabel.textColor = .simplenoteNoteHeadlineColor
+    }
+
+    func styleErrorMessageLabel() {
+        errorMessageLabel.textColor = .simplenoteTextColor
+    }
+
+    func styleSlider() {
+        let color = UIColor.simplenoteGray50Color.withAlphaComponent(0.2)
+        slider.minimumTrackTintColor = color
+        slider.maximumTrackTintColor = color
+    }
+
+    func styleRestoreButton() {
+        restoreButton.backgroundColor = restoreButton.isEnabled ? .simplenoteBlue50Color : .simplenoteDisabledButtonBackgroundColor
+        restoreButton.setTitle(NSLocalizedString("Restore Note", comment: "Restore a note to a previous version"), for: .normal)
+    }
+
+    func styleActivityIndicator() {
+        if #available(iOS 13.0, *) {
+            activityIndicator.style = .medium
+        } else {
+            activityIndicator.style = .gray
+        }
+    }
+
+    func update(with state: SPNoteHistoryController.State) {
+        switch state {
+        case .loading:
+            setMainContentVisible(false)
+            setActivityIndicatorVisible(true)
+            setErrorMessageVisible(false)
+
+        case .results(let items):
+            setMainContentVisible(true)
+            setActivityIndicatorVisible(false)
+            setErrorMessageVisible(false)
+
+            self.items = items
+
+            configureSlider()
+
+        case .error(let text):
+            setMainContentVisible(false)
+            setActivityIndicatorVisible(false)
+            setErrorMessageVisible(true)
+
+            errorMessageLabel.text = text
+        }
+    }
+
+    func update(withSliderValue value: Float) {
+        let index = Int(value)
+        let item = items[index]
+
+        dateLabel.text = item.date
```

Comment:
At least in the Electron app we've had a lot of confusion relating to the display of the date for revisions. While we're in here we might want to think about clearing some of that up. cc: @SylvesterWilmott 

In the work I've been doing, for example, I show something like "Originally created: June 30, 2020 8:31 PM" to show that the revisions don't follow a linear ordering in time so much as a linear sequence of edits which may have happened out-of-order. This helps too when sliding back over restored versions, showing that they might have been a restore from a much earlier version of the note.

Reply:
Interesting! 
On iOS `modificationDate` is shown when user picks a version and they seem to be in linear order.
If I restore a note to some previous version and open history again, the last entry will have a current date. I also noticed that `creationDate` for each version is the same...

Something worth investigating?

Reply:
> they seem to be in linear order.

This is a source of confusion because _they aren't_ 😉 

The only way to linearize the versions is to list them in the order that the server accepted them. This is actually independent from the `modificationDate` and `creationDate`.

`creationDate` should never change because it's an intrinsic property of the note throughout its lifetime.

`modificationDate` is something each device sets and should do that whenever edits are made on that device. They can be out of order in the history. For example, open a note on device A and on device B. With device B offline make an edit to the note. Then, _after_ that make an edit to the same note on device A. Device A's change will hit the server first. Bring device B back online and its change will hit the server second but the `modificationDate` timestamps will now be backwards.

I may not have realized that iOS is changing the modification date when restoring but personally I like leaving it in tact _so that_ we can show "this version of the note history was a revert from some previous point in time and this is that point in time."

It's all confusing; acting like it's a chain of edits in `modificationDate` time order though is one of the most-confusing ways to represent it.

Reply:
Thanks for the explanation! 
Behavior is quite confusing indeed, especially if different platforms handle it slightly different 🤔